dot/core/sync.go

```diff
@@ -257,7 +263,33 @@ func (s *Syncer) handleBlockResponse(msg *network.BlockResponseMessage) (int64,
 				Body:   body,
 			}
 
-			// TODO: execute block and verify authorship right
+			// prepare block for sending to core_executeBlock,
+			//  core_executeBlock fails if Digest and Body data are sent
+			blockData := types.Block{
+				Header: &types.Header{
+					ParentHash:     header.ParentHash,
+					Number:         header.Number,
+					StateRoot:      header.StateRoot,
+					ExtrinsicsRoot: header.ExtrinsicsRoot,
+				},
+				Body: types.NewBody([]byte{}),
+			}
+
+			bdEnc, err := blockData.Encode()
+			if err != nil {
+				return 0, err
+			}
```

Comment:
could you move this logic to the `executeBlock` function, and have `executeBlock` take a block instead?

Reply:
done.